@types/redislock.d.ts

```diff
@@ -0,0 +1,36 @@
+import { Redis } from "ioredis";
+export {
+  LockAcquisitionError,
+  LockExtendError,
+  LockReleaseError
+} from "../src/errors";
+export import Lock = require('../src/lock');
+
+/** redislock offers both atomic acquire and release operations,
+ * avoiding race conditions among clients, as well as the need for lock-specific redis connections.
+ * Lock creation requires a node_redis client,
+ * and accepts an object specifying the following three options: */
+export interface LockOption {
+  /** timeout Time in milliseconds before which a lock expires
+   * (default: 10000 ms) */
+  timeout: number;
+  /** retries Maximum number of retries in acquiring a lock if the first attempt failed
+   * (default: 0) */
+  retries: number;
+  /** delay   Time in milliseconds to wait between each attempt
+   * (default: 50 ms) */
+  delay: number;
+}
+
+/** Creates and returns a new Lock instance,
+ * configured for use with the supplied redis client, as well as options,
+ * if provided. The options object may contain following three keys,
+ * as outlined at the start of the documentation: timeout, retries and delay. */
+export function createLock(client: Redis, option: LockOption): Lock;
```

Comment:
should be Redis or Redis.Cluster

Reply:
new Redis(); return Redis.Redis not Redis.Client and there are not exported Redis.Client

refs: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/516777f11e14552e61a7ca002ad03f72d69e96fb/types/ioredis/index.d.ts#L34

![Screenshot from 2020-10-28 19-59-12](https://user-images.githubusercontent.com/13000056/97439171-6c510e80-1958-11eb-8e13-fa3293c996a1.png)

Reply:
Cluster not client :)

Reply:
Oh my bad, fixed.